src/components/Header/Header.tsx

```diff
@@ -0,0 +1,92 @@
+import React, {
+  FC, useCallback, useMemo, useState,
+} from 'react';
+import cn from 'classnames';
+import { Errors } from '../../types/Errors';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todos: Todo[],
+  setErrorMessage: (error: Errors) => void,
+  addTodo: (inputValue: string) => Promise<void>,
+  updateTodo: (updatedTodo: Todo) => Promise<void>,
+};
+export const Header: FC<Props> = (props) => {
+  const {
+    todos,
+    setErrorMessage,
+    addTodo,
+    updateTodo,
+  } = props;
+
+  const [inputValue, setInputValue] = useState('');
+  const [isDisabledInput, setIsDisabledInput] = useState(false);
+  const handleFocusOnInput = useCallback((input: HTMLInputElement) => {
+    input?.focus();
+  }, []);
+
+  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!inputValue.trim()) {
+      setErrorMessage(Errors.EmptyTitle);
+      setTimeout(() => setErrorMessage(Errors.Null), 3000);
+
+      return;
+    }
+
+    setErrorMessage(Errors.Null);
+    setIsDisabledInput(true);
+
+    addTodo(inputValue)
+      .then(() => {
+        setInputValue('');
+      })
+      .finally(() => setIsDisabledInput(false));
+  };
+
+  const completeAllTodos = async () => {
+    const statusOfTodos = [...todos].every(todo => todo.completed);
+
+    const updateTodos = statusOfTodos
+      ? todos.filter(todo => todo.completed)
+      : todos.filter(todo => !todo.completed);
```

Comment:
```suggestion
    const updateTodos = todos.filter(todo => statusOfTodos);
```
Looks like you can simplify